enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,504 @@
+
+### Set a broader policy for more Red Hat images
+
+I think a policy for all of `quay.io/openshift-release-dev` (both the `quay.io/openshift-release-dev/ocp-release` release images and the `quay.io/openshift-release-dev/ocp-v4.0-art-dev` images referenced from release images), would provide more protection.
+Covering the operator-lifecycle manager (OLM) catalogs in `registry.redhat.io/redhat/*-operator-index` and other Red Hat images would provide even more protection.
+But I'm limiting this initial proposal to just `quay.io/openshift-release-dev/ocp-release` to get the smallest possible slice of progress.
+If the initial steps go well, we can extend the image policy to other Red Hat scopes in future iterations.
+
+### Use an ImagePolicy for `openshift-*` namespaces
+
+The current ClusterImagePolicy approach restricts the scoped images, regardless of which namespace they run in.
+But it does not restrict other images from running in `openshift-*` namespaces, including `openshift-cluster-version`.
+Ideally we would use ImagePolicy as well to add those restrictions, but again, I'm looking for the smallest possible slice of progress.
+And while [api#1457][api-1457] landed both ClusterImagePolicy and ImagePolicy, [mco#4160][machine-config-operator-4160] only pulled the former into the cluster.
+We might come back and pick up namespaced policies once [ImagePolicy has been implemented][OCPNODE-2333], but I'm deferring on OpenShift policies based on that custom resource for now.
+
+### Delay running a cluster-version operator pod until the image policy is in place
+
+We could add logic to ensure the current, relevant image policies were in place before launching a new release image pod.
+This would avoid gaps where a cluster-version operator (CVO) pod, or [the `version-*` pod it uses to get manifests for an incoming release image][cluster-version-operator-payload-job], launch against old policies (or on install, against a non-existent policy).
+
+But while there is some has been applied wording in [the (Cluster)ImagePolicy enhancement](/enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md), there are some corner cases, like one compute machine not letting us roll out the policy, that probably shouldn't block CVO pod creation.
+As it stands, this proposal lets the new `quay.io/openshift-release-dev/ocp-release` Pods race ClusterImagePolicy.
+If the images violate the incoming image policy, they'll fail on their first reschedule.
+It's not clear from [the (Cluster)ImagePolicy enhancement](/enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md) whether a policy change triggers the drain/replacement or other conformance checks for existing pods/containers running images in the scopes whose policy is changing.
```

Comment:
FYI policy change (/etc/containers/policy.json change) does not trigger the node drain. It runs a [crio reload](https://github.com/openshift/machine-config-operator/blob/ff7ef2ec8ddbdf4f5758ee8f3ba3fea2d364e581/docs/MachineConfigDaemon.md?plain=1#L165).

Reply:
Thanks for the link!  I've pushed ecade8dc -> b9c36410, replacing this sentence with:

```Markdown
(Cluster)ImagePolicy changes will [trigger CRI-O reloads][machine-config-operator-reload-crio], but not drain the node, so there might not be any immediate sign that a policy change had broken the ability to pull a critical image.
```

I'll mark this thread resolved, but let me know if you think I still need text chances to capture this distinction :)